internal/telemetry/telemetry_test.go

```diff
@@ -19,35 +19,39 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestProductChange(t *testing.T) {
+func TestProductEnabled(t *testing.T) {
 	client := new(client)
 	client.start(nil, NamespaceTracers)
-	client.productChange(NamespaceProfilers, true,
-		[]Configuration{BoolConfig("delta_profiles", true)})
-
+	client.productEnabled(NamespaceProfilers)
 	// should contain app-client-configuration-change and app-product-change
-	assert.Len(t, client.requests, 2)
+	assert.Len(t, client.requests, 1)
```

Comment:
A few of these length checks can be switched to `require` for slightly better failing test output